src/main/kotlin/racingcar/RacingGame.kt

```diff
@@ -0,0 +1,51 @@
+package racingcar
+
+import racingcar.domain.Car
+import racingcar.domain.Cars
+import racingcar.domain.Count
+import racingcar.view.*
+
+class RacingGame {
+
+    fun play() {
+        showNameInputMessage()
+        val cars = tryMakeCars()
+
+        showCountInputMessage()
+        val count: Count = tryMakeCount()
+
+        val winners = playRace(cars, count)
+
+        showWinners(winners)
+    }
+
+    private fun tryMakeCars(): Cars {
+        return try {
+            makeCars()
+        } catch (e: IllegalArgumentException) {
+            println(e.message)
+            tryMakeCars()
+        }
+    }
+
+    private fun makeCars() = Cars(readNames().map { Car(it) })
+
+    private fun tryMakeCount(): Count {
+        return try {
+            makeCount()
+        } catch (e: IllegalArgumentException) {
+            println(e.message)
+            tryMakeCount()
+        }
+    }
+
+    private fun makeCount() = Count(readln().toIntOrNull() ?: throw IllegalArgumentException("숫자를 입력해주세요."))
+
+    private fun playRace(cars: Cars, count: Count): List<Car> {
+        for (i in 1..count.value) {
+            cars.moveAll()
+            showCarsPosition(cars)
+        }
```

Comment:
`repeat`을 사용할 수도 있을 것 같아요 :)

```suggestion
        repeat(count.value) {
            cars.moveAll()
            showCarsPosition(cars)
        }
```

Reply:
`for (i in 1..n)`도 신기했는데 이건 더 신기하네요..!
코틀린은 한 기능에도 정말 많은 키워드가 있는 것 같아요, 익숙한 걸 쓰기 전에 더 찾아봐야겠습니다 👍